src/status_im/chat/models/reactions.cljs

```diff
@@ -105,3 +106,39 @@
        acc))
    []
    reactions))
+
+(defn- <-rpc
+  [emoji-reaction]
+  (-> emoji-reaction
+      (set/rename-keys
+       {:emojiId       :emoji-id
+        :compressedKey :compressed-key})
+      (select-keys [:emoji-id :from :compressed-key])))
+
+(defn- format-response
+  [response]
+  (->> (js->clj response :keywordize-keys true)
+       (map <-rpc)
+       (group-by :emoji-id)
+       (into (sorted-map))))
+
+(rf/defn save-emoji-reaction-details
+  {:events [:chat/save-emoji-reaction-details]}
+  [{:keys [db]} message-reactions]
+  {:db (assoc-in db [:chats (:current-chat-id db) :message/reaction-list] message-reactions)})
```

Comment:
It *might* be a good idea to check if `(:current-chat-id db)` returns something before using it as a key. Kind of a "validate stuff before writing to the db" approach. But I'm probably over worried.

Reply:
I think it's not possible to dispatch this event from outside a chat, As reactions are only in chats for now.